bgpd/bgp_route.c

```diff
@@ -10223,17 +10224,29 @@ static void show_adj_route(struct vty *vty, struct peer *peer, afi_t afi,
 				}
 
 				bgp_attr_dup(&attr, ain->attr);
-				ret = bgp_input_modifier(peer, &rn->p, &attr,
-							 afi, safi, rmap_name);
+				route_filtered = false;
+
+				/* Filter prefix using distribute list,
+				 * filter list or prefix list
+				 */
+				if ((bgp_input_filter(peer, &rn->p, &attr, afi,
+						safi)) == FILTER_DENY)
+					route_filtered = true;
+
+				/* Filter prefix using route-map */
+				if ((bgp_input_modifier(peer, &rn->p, &attr,
+							afi, safi, rmap_name))
+					== RMAP_DENY)
+					route_filtered = true;
 
 				if (type == bgp_show_adj_route_filtered
-				    && ret != RMAP_DENY) {
+				    && !route_filtered) {
```

Comment:
Use the route_filtered condition only to capture the result of bgp_input_filter(). Keep the RMAP_DENY condition as it.